[AN] feat: 상세 화면에서 카테고리, 키워드 클릭 시 화면 전환

HamBeomJoon opened the pull request:
<!-- PR 제목 예시
ex. [BE] 로그인 API 구현
ex. [AN] 페이지 구현 -->

## 📌 변경 내용 & 이유
<!-- ex. 
- 사용자 로그인 기능 구현
- 로그인 실패 시 에러 메시지 반환 처리 추가 -->
- 상세 화면에서 카테고리, 키워드 클릭 시 화면 전환

## 📸 스크린샷 (선택)
> UI 변경이 있을 경우 첨부

## 🧪 테스트 방법 (선택)
<!-- 코드 리뷰하는 팀원이 돌려볼 수 있도록 작성해주기 
ex. ReservationTest의 XXXX 테스트메서드로 확인 
ex. 어떤화면의 어떤기능에서 확인 -->

## 📢 논의하고 싶은 내용
<!-- 이러이러해서 이부분을 이러이러하게 구현했는데 이방식이 괜찮은지? -->

## 🧩 관련 이슈
<!-- 이슈 태그: #123 -->
<!-- 이슈 닫기: close #123 -->
#443 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신기능**
  * 플레이어 상세에서 카테고리/키워드 클릭 시 검색 탭으로 이동해 즉시 결과 표시
  * 설정에 "피드백 및 문의하기" 항목 추가(웹폼으로 이동)
  * 드로어에 이용약관 바로가기 추가

* **개선**
  * 상세/탐색/라이브러리 간 이동을 Activity Result 기반으로 정비해 흐름 안정화
  * 검색 자동 진입을 최초 생성 시로만 제한
  * 키워드·카테고리 선택 시 사용자 행동 로깅 강화

* **버전**
  * 앱 버전 1.1.1로 업데이트
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

HamBeomJoon committed:
[AN] feat: 상세 화면에서 카테고리 클릭 시 카테고리 검색 화면으로 이동 #443

HamBeomJoon committed:
[AN] feat: 탐색 화면에서 상세 화면 진입 후 다른 화면으로 이동 시 분기 처리 #443

HamBeomJoon committed:
[AN] feat: 라이브러리 화면에서 상세 화면 진입 후 다른 화면으로 이동 시 분기 처리 #443

HamBeomJoon committed:
[AN] feat: 하단 재생 바에서 상세 화면 진입 후 다른 화면으로 이동 시 분기 처리 #443

HamBeomJoon committed:
[AN] build: v1.1.1 버전 업그레이드 #443

rosemin928 commented on android/app/src/main/java/com/onair/hearit/presentation/explore/ExploreFragment.kt:
```diff
@@ -49,17 +53,36 @@ class ExploreFragment :
 
     private val playerDetailLauncher =
```
이 부분 코드가 너무 길어서 가독성이 살짝 떨어지는 것 같아요...!
안에 들어가는 로직을 함수로 분리해봐도 좋을 것 같습니당
그리고 `type`도 상수로 관리하거나, 할 수 있다면 `enum class` 같은 것들로 관리하도록 하면 더 좋을 것 같아욤

HamBeomJoon replied:
```
sealed class DetailResult {
    data class Explore(
        val hearitId: Long,
        val bookmarkId: Long?,
    ) : DetailResult()

    data class Category(
        val bundle: Bundle,
    ) : DetailResult()

    data class Keyword(
        val bundle: Bundle,
    ) : DetailResult()
}
```
DetailResult를 sealed class로 관리하고 확장 함수로 분리를 해봤습니다

rosemin928 reviewed (approved):
카테고리, 키워드 화면 전환 수고하셨습니다~!
코멘트 하나 남겨뒀는데 확인해주세요!

HamBeomJoon committed:
[AN] refactor: 초기 진입 시: 기본 화면 구성 후 필요하면 즉시 결과로 이동 #443

gahyunkim commented on android/app/src/main/java/com/onair/hearit/presentation/MainActivity.kt:
```diff
@@ -90,8 +91,13 @@ class MainActivity :
+                    val extras = data.extras ?: return@registerForActivityResult
+                    navigateToSearchResult(extras)
+                } else {
+                    Timber.d("Detail closed with no search input")
```
Timber는 지워주시면 좋을 것 같아요!

gahyunkim reviewed (approved):
수고하셨습니다~!

HamBeomJoon committed:
[AN] refactor: Intent Keys, Values 파일 분리 #443

HamBeomJoon committed:
[AN] feat: DetailActivity 결과 처리 개선 #443

- `DetailResult` Sealed Class 정의: Explore, Category, Keyword 타입을 포함하여 DetailActivity의 다양한 결과 유형을 나타냅니다.
- `Intent.toDetailResult()` 확장 함수 추가: Intent에서 `DetailResult` 객체를 생성합니다.
- `DetailResult.navigate()` 확장 함수 추가: `DetailResult` 유형에 따라 MainActivity에서 적절한 화면 이동을 처리합니다.

HamBeomJoon committed:
[AN] feat: 피드백 채널 추가 #443

HamBeomJoon committed:
Merge branch 'develop-an' into feat/#443-detail-goto-search

HamBeomJoon committed:
[AN] chore: 기타 수정 #443
